autoextract/aio/errors.py

```diff
@@ -15,3 +22,64 @@ def __str__(self):
         return f"ApiError: {self.status}, message={self.message}, " \
                f"headers={self.headers}, body={self.response_content}"
 
+
+class QueryError(Exception):
+    """ Exception which is raised when a Query-level error is returned.
+    https://doc.scrapinghub.com/autoextract.html#query-level
+    """
+
+    DOMAIN_OCCUPIED_REGEX = re.compile(
+        r".*domain (.+) is occupied, please retry in (.+) seconds.*",
+        re.IGNORECASE
+    )
+
+    RETRIABLE_QUERY_ERROR_MESSAGES = {
+        msg.lower().strip()
+        for msg in [
+            "query timed out",
+            "Downloader error: No response (network5)",
+            "Downloader error: http50",
+            "Downloader error: GlobalTimeoutError",
+            "Proxy error: banned",
+            "Proxy error: internal_error",
+            "Proxy error: nxdomain",
+            "Proxy error: timeout",
+            "Proxy error: ssl_tunnel_error",
+            "Proxy error: msgtimeout",
+            "Proxy error: econnrefused",
+            # Retry errors for AutoExtract API dev server
+            "Error making splash request: ServerDisconnectedError",
+            "Error making splash request: ClientOSError: [Errno 32] Broken pipe",
+        ]
+    }
+
+    def __init__(self, query: dict, message: str):
+        self.query = query
+        self.message = message
+
+    def __str__(self):
+        return f"QueryError: query={self.query}, message={self.message}"
+
+    @property
+    def retriable(self) -> bool:
+        if self.domain_occupied:
+            return True
+
+        return self.message.lower() in self.RETRIABLE_QUERY_ERROR_MESSAGES
+
+    @property
+    def domain_occupied(self) -> Optional[str]:
```

Comment:
It seems it's used only as a `bool` result, so you can update the typing to be `bool` and return the proper value

Reply:
Not anymore :)